R/ncaa_schedule.R

```diff
@@ -0,0 +1,243 @@
+#' @title ncaa_schedule
+#'
+#' @description Scrapes the schedule for a given team from the NCAA. Returns data similar to
+#' `bigballR` but does so more efficiently and quickly.
+#'
+#' @param team_id the team ID for a given team
+#' @return a data frame with the following columns: date, game_id, home_team, home_team_id,
+#' away_team, away_team_id, home_score, away_score, is_neutral, details
+#'
+#' @examples
+#' \donttest{
+#' ncaa_schedule('501404') # womens soccer
+#' ncaa_schedule('509092') # baseball
+#' ncaa_schedule('509675') # mens tennis
+#' }
+#'
+#' @export
+ncaa_schedule <- function(team_id) {
+  html_a <- glue::glue('https://stats.ncaa.org/teams/{team_id}') |>
+    rvest::read_html()
+  game_by_game_url <- html_a |>
+    rvest::html_node('#contentarea') |>
+    rvest::html_nodes('a') |>
+    rvest::html_attr('href') |>
+    (\(x) {
+      x[stringi::stri_detect(x, regex = 'player/game_by_game')]
+    })() |>
+    dplyr::first()
+  html_b <-
+    glue::glue('https://stats.ncaa.org/{game_by_game_url}') |>
+    rvest::read_html()
+  if(html_b |>
+     rvest::html_text() != 'Unable to find player'){
+    team_name <- html_b |>
+      rvest::html_node('body') |>
+      rvest::html_node('#contentarea') |>
+      rvest::html_node('#stats_player_person_id') |>
+      rvest::html_node('option') |>
+      rvest::html_text()
+    game_id_table <- data.frame(
+      Date = html_b |>
+        rvest::html_node('body') |>
+        rvest::html_node('div#contentarea') |>
+        rvest::html_node('div#game_breakdown_div') |>
+        rvest::html_nodes(xpath='table/tr/td/table/tr/td[count(table/tr/td/table/tr/th[.="Result"]/preceding-sibling::th)+1]') |>
+        rvest::html_text(trim=T) |>
+        unlist() |>
+        (\(x) x[x!=''])() |>
+        (\(x) x[-c(1)])(),
+      Opponent_Clean = html_b |>
+        rvest::html_node('body') |>
+        rvest::html_node('div#contentarea') |>
+        rvest::html_node('div#game_breakdown_div') |>
+        rvest::html_nodes(xpath='table/tr/td/table/tr/td[count(table/tr/td/table/tr/th[.="Result"]/preceding-sibling::th)+2]') |>
+        lapply(\(x) {
+          if(length(rvest::html_attr(x,'class'))){
+            if(!is.na(rvest::html_attr(x,'class'))){
+              return(rvest::html_text(x,trim=T))
+            }
+          }}) |>
+        unlist() |>
+        stringi::stri_replace_all(regex = '((?=(.*)) @ (.*)$)|(@ )', ''),
```

Comment:
How are we handling cancelled/postponed games here?

Reply:
Believe that the left join in line 205 _does not_ filter cancelled/postponed games, you just scores/results listed as NA? should probably be an inner_join instead if we want to filter them.